app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, inverse_of: 'post', dependent: :destroy
+  has_many :likes, inverse_of: 'post', dependent: :destroy
+
+  def recent_five_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_post_counter
+    author.increment!(:posts_count)
+  end
```

Comment:
- [ ] I think you are trying to use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) here where you can use `after_save` to call the #update_post_counter method every time a new post has been saved to a database. I said that because I can see your method is in private. Since you've made the method private, you need to add the `after_save :update_post_counter` for the Active Callbacks to be perfected.